crates/miden-tx/src/auth/tx_authenticator.rs

```diff
@@ -1,15 +1,55 @@
 use alloc::{collections::BTreeMap, string::ToString, sync::Arc, vec::Vec};
 
 use miden_objects::{
-    Felt, Word,
-    account::{AccountDelta, AuthSecretKey},
-    utils::sync::RwLock,
+    Felt, Hasher, Word, account::AuthSecretKey, crypto::SequentialCommit,
+    transaction::TransactionSummary, utils::sync::RwLock,
 };
 use rand::Rng;
 
 use super::signatures::get_falcon_signature;
 use crate::errors::AuthenticationError;
 
+// SIGNATURE DATA
+// ================================================================================================
+
+/// Data types on which a signature can be requested.
+///
+/// It supports three modes:
+/// - `TransactionSummary`: Structured transaction summary, recommended for authenticating
+///   transactions.
+/// - `Arbitrary`: Arbitrary payload provided by the application.
+/// - `Blind`: The underlying data is not meant to be displayed in a human-readable format. It can
+///   be assumed to already be a cryptographic commitment to some data.
+#[allow(clippy::large_enum_variant)]
+pub enum SignatureData {
+    TransactionSummary(TransactionSummary),
+    Arbitrary(Vec<Felt>),
+    Blind(Word),
+}
```

Comment:
Nit: Not critical, but I think I would follow the lint suggestion and `Box` the `TransactionSummary` since we otherwise allocate ~260 bytes on the stack even when using `Blind(Word)`, wdyt?

Reply:
Yeah, I think this is a case-by-case thing, since it really depends on how it is used, but there are probably other ones where we could also reconsider our approach.

Reply:
Applied in 25eabbfd